source/Missions/Services/Arena/CoopArenaController.cs

```diff
@@ -1,158 +1,264 @@
+            Vec2 vec = frame.rotation.f.AsVec2;
+            vec = vec.Normalized();
+            Agent agent = mission.SpawnAgent(agentBuildData.InitialDirection(vec)
+                .NoHorses(true)
+                .Equipment(CreateRandomEquipment(false))
+                .TroopOrigin(new SimpleAgentOrigin(character, -1, null, default)), false, 0);
+            agent.FadeIn();
+            agent.Controller = Agent.ControllerType.Player;
+            return agent;
+        }
+
+        public Agent SpawnAgent(Vec3 startingPos, CharacterObject character)
+        {
+            AgentBuildData agentBuildData = new AgentBuildData(character);
+            agentBuildData.BodyProperties(character.GetBodyPropertiesMax());
+            agentBuildData.InitialPosition(startingPos);
+            agentBuildData.Team(Mission.Current.PlayerAllyTeam);
+            agentBuildData.InitialDirection(Vec2.Forward);
+            agentBuildData.NoHorses(true);
+            agentBuildData.Equipment(CreateRandomEquipment(false));
+            agentBuildData.TroopOrigin(new SimpleAgentOrigin(character, -1, null, default));
+            agentBuildData.Controller(Agent.ControllerType.None);
+
+            Agent agent = default;
+            GameLoopRunner.RunOnMainThread(() =>
+            {
+                agent = Mission.Current.SpawnAgent(agentBuildData);
+                agent.FadeIn();
+            });
+
+            return agent;
+        }
+
+        // DEBUG METHOD: Starts an Arena fight
+        public void StartArenaFight()
+        {
+            // reset teams if any exists
+            Mission.Current.ResetMission();
+
+            Mission.Current.Teams.Add(BattleSideEnum.Defender, Hero.MainHero.MapFaction.Color, Hero.MainHero.MapFaction.Color2, null, true, false, true);
+            Mission.Current.Teams.Add(BattleSideEnum.Attacker, Hero.MainHero.MapFaction.Color2, Hero.MainHero.MapFaction.Color, null, true, false, true);
+
+            // players is defender team
+            Mission.Current.PlayerTeam = Mission.Current.DefenderTeam;
+
+
+            // find areas of spawn
+            List<MatrixFrame> spawnFrames = (from e in Mission.Current.Scene.FindEntitiesWithTag("sp_arena")
+                                             select e.GetGlobalFrame()).ToList();
+            for (int i = 0; i < spawnFrames.Count; i++)
+            {
+                MatrixFrame value = spawnFrames[i];
+                value.rotation.OrthonormalizeAccordingToForwardAndKeepUpAsZAxis();
+                spawnFrames[i] = value;
+            }
+            // get a random spawn point
+            MatrixFrame randomElement = spawnFrames.GetRandomElement();
+
+
+            // spawn an instance of the player (controlled by default)
+            SpawnPlayerAgent(CharacterObject.PlayerCharacter, randomElement);
+        }
+        
+        //creates random equipment for characters

```

Comment:
Convert all comments above the new methods to xml comments. Also add comments for any parameters
If you remove the current comment and type `///` Visual Studio will template it automatically
```suggestion
        /// <summary>
        /// Creates random equipment for characters
        /// </summary>
```